Library/Homebrew/cmd/bottle.rb

```diff
@@ -183,6 +186,15 @@ def bottle_formula(f)
     relocatable = false
     skip_relocation = false
 
+    formula_source_time = f.stable.stage do
+      newest_mtime = nil
+      Pathname(Dir.pwd).find do |path|
+        newest_mtime ||= path.mtime
+        newest_mtime = path.mtime if path.mtime > newest_mtime
+      end
+      newest_mtime
+    end
```

Comment:
I'd rather the performance penalty be in `bottle` rather than `install` as there's a lot more `install`s happening than `bottle`s.